### PR TITLE
chore(telemetry): generate tracking plan from types COMPASS-8058

### PR DESCRIPTION
**Pull request by mcasimir**
Generates a tracking plan in `docs/tracking-plan.md` from `telemetry-events.ts` along the other auto-generated files.

### CODE REVIEW

File: packages/compass-telemetry/src/telemetry-events.ts [hunk 1]
```diff
@@ -79,7 +82,7 @@ export type ConnectionScopedProperties = {
   /**
    * The id of the connection associated to this event.
    */
-  connection_id: string;
+  connection_id?: string;
```
Comment (mcasimir) [inline packages/compass-telemetry/src/telemetry-events.ts:85]: looks like `connection_id` is not guaranteed for all the connection scoped events

File: scripts/generate-tracking-plan.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+  const program = ts.createProgram(['inMemoryFile.ts'], compilerOptions, host);
+  const checker = program.getTypeChecker();
+  return { sourceFile, checker };
+}
+
+function generateMarkdownPlan(
+  events: TelemetryEventInfo[],
+  identifyTraits: TelemetryEventInfo
+) {
+  const byCategory = _.groupBy(events, 'category');
```
Comment (Anemy) [inline scripts/generate-tracking-plan.ts:266]: Totaly nit, since we're only using lodash for this can we do it in regular ts and avoid the dependency?

**Review by Anemy — APPROVED** [review]

File: scripts/generate-tracking-plan.ts (hunk 2 above)
Comment (mcasimir) [reply scripts/generate-tracking-plan.ts]: True, removed and also sorted everything

File: packages/compass-telemetry/src/telemetry-events.ts (hunk 1 above)
Comment (lerouxb) [reply packages/compass-telemetry/src/telemetry-events.ts]: Any idea which ones? Sounds dodgy.

Comment (mcasimir) [reply packages/compass-telemetry/src/telemetry-events.ts]: Screen is one that is hard to fix, and then we found some others when connecting from CLI without an existing id

**Review by lerouxb — APPROVED** [review]

**Pull request merged**